compiler/modules/print.wasm/binaryen.bal

```diff
@@ -0,0 +1,74 @@
+
+
+public type IntType "i64"|"i32";
+
+public type Type "None"|IntType;
+
+class Module {
+    private Function[] functions;
+    private map<Function> functionMap;
+    private Export[] exports;
+    private map<Export> exportMap;
+
+    function call(string target, Expression[] operands, int numOperands, Type returnType) returns Expression {
+        panic error("unimplemented");
+    }
+
+    function localGet(int index, Type ty) returns Expression {
+        panic error("unimplemented");
+    }
+    
+    function addConst(Literal value) returns Expression {
+        panic error("unimplemented");
+    }
+
+    function addReturn(Expression? value= ()) returns Expression {
+        panic error("unimplemented");        
+    }
+
+    function nop() returns Expression {
+        panic error("unimplemented");        
+    }
+         
+    function addFunction(string name, Type[] params, Type results, Type[] varTypes, int numVarTypes, Expression body)  {
+        panic error("unimplemented");        
+    }
+
+    function addFunctionImport(string internalName, string externalModuleName, string externalBaseName, Type params, Type results)  {
+        panic error("unimplemented");        
+    }
+
+    function addFunctionExport(string internalName, string externalName) {
+        panic error("unimplemented");        
+    }
+
+    //BinaryenModuleDispose and BinaryenModulePrint
+    function finish(){
+        panic error("unimplemented");
+    }
+                                               
+}
+
+
+type Function record {
+
+};
+
+type Export record {
+     
+};
+
+type Expression record {
+    
+};
+
+
+type LiteralInt32 record {
+    int value;
+};
+
+type LiteralInt64 record {
+    int value;
+};
+
+type Literal LiteralInt32|LiteralInt64;
```

Comment:
Ballerina is a structurally typed language, what that means is, it can't tell the difference between ` LiteralInt32` and `LiteralInt64`, eg if `x is LiteralInt32` is `true`, `x is LiteralInt64` is also `true`.
There are two ways to get around the issue,
1) Make them field names different
```ballerina
type LiteralInt32 record {
    int i32;
};
```
2) Add a consent field. (See `bir.bal` for a real example)
```ballerina
type LiteralInt32 record {
    "i32" type = "i32";
    int value;
};
```
I am not 100%  sure what will turn out best, let's go with 1 and see.